juju/juju
# Migrate storage docs to use hostid not machineid

## Description of change

The 4 collections used to store volume and filesystem info are migrated to use "hostid" instead of "machineid". The bulk of the change is to write the upgrade step. There was an existing 2.2 upgrade step that needed to be modified to use bson maps instead of juju state structs (arguably all upgrade steps should be written this way anyway).

## QA steps

Deploy Juju 2.1 with postgresql using storage
Upgrade to 2.5
Check db to ensure migrations occurred
Check storage functionality
